dashboard/src/main/java/com/google/cloud/tools/opensource/cloudbomdashboard/VersionData.java

```diff
@@ -0,0 +1,268 @@
+        if (artifactToLatestVersion.containsKey(artifact)) {
+            return artifactToLatestVersion.get(artifact);
+        }
+        String pomPath = getMetadataURL(artifact);
+        try {
+            URL url = new URL(pomPath);
+            URLConnection conn = url.openConnection();
+            conn.setConnectTimeout(2000);
+            conn.setReadTimeout(2000);
+            Scanner s = new Scanner(conn.getInputStream());
+            while (s.hasNextLine()) {
+                String string = s.nextLine();
+                if (string.contains("<latest>")) {
+                    String version = string.split(">")[1].split("<")[0];
+                    artifactToLatestVersion.put(artifact, version);
+                    return version;
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        artifactToLatestVersion.put(artifact, "");
+        return "";
+    }
+
+    private static String updatedTime(Artifact artifact) {
+        if (artifactToTime.containsKey(artifact)) {
+            return artifactToTime.get(artifact);
+        }
+        String groupPath = artifact.getGroupId().replace('.', '/');
+        String metadataPath = DashboardMain.basePath + "/" + groupPath
+                + "/" + artifact.getArtifactId()
+                + "/maven-metadata.xml";
+        try {
+            URL url = new URL(metadataPath);
+            URLConnection conn = url.openConnection();
+            conn.setConnectTimeout(2000);
+            conn.setReadTimeout(2000);
+            Scanner s = new Scanner(conn.getInputStream());
+            while (s.hasNextLine()) {
+                String string = s.nextLine();
+                if (string.contains("<lastUpdated>")) {
+                    DateFormat dateFormat = new SimpleDateFormat("yyyyMMddhhmmss");
+                    DateFormat outputFormat = new SimpleDateFormat("MM-dd-yyyy");
+                    String input = string.split(">")[1].split("<")[0];
+                    Date date = dateFormat.parse(input);
+                    artifactToTime.put(artifact, outputFormat.format(date));
+                    return outputFormat.format(date);
+                }
+            }
+        } catch (IOException | java.text.ParseException e) {
+            e.printStackTrace();
+        }
+        artifactToTime.put(artifact, "");
+        return "";
+    }
+
+    /**
+     * @param key                Key to use when inserting the artifact's associated path into the given map.
+     * @param artifact           Artifact to add into the map
+     * @param sharedDepsPosition The map receiving the path associated with this artifact.
+     * @return Returns the version of shared-dependencies if found. Returns the empty string otherwise.
+     */
+    private static String sharedDependencyVersion(String key, org.eclipse.aether.artifact.Artifact artifact, Map<String, String> sharedDepsPosition) {
```

Comment:
sharedDepsPosition --> sharedDependencyPosition (google style does not abbreviate in variable names)